File: erp/src/types/Product.ts

export interface Variant{
  color?: string
  size: string
  quantity: number
  cost: number
  price: number
  discount?: number
  images: string[]
}

export interface Product{
  name: string
  description?: string
  image: string
  category_ids?: number[]
  variants: Variant[]

}

File: server/app/api/schemas/product.py
from typing import Optional, List, TYPE_CHECKING
from sqlmodel import SQLModel, Field, Relationship

class Image(SQLModel):
    url: str
    name: str
    is_principal: Optional[bool] = False

class SizeCreate(SQLModel):
    name: str

class Option(SQLModel):
    size: str
    quantity: int

class ProductSchema(SQLModel):
    name: str
    description: Optional[str]
    cost: float
    price: float
    images: List[Image]
    options: List[Option]

